CDCgov/prime-reportstream
# 12722: Enrich logs with additional queryable data

This PR adds structured logging in JSON to Report Stream. This allows Azure AppInsights to be able to pull apart different aspects of the logs into queryable data. The AppInsights Java agent will also inject additional fields (such as a trace id!) in the MDC.

This PR also adds helpful logging extension functions to able to take advantage of adding fields to the MDC. This allows us to query for logs based on custom fields defined in our source code. The fields will show up in the `customDimensions` property in the `traces` table

Test Steps:
1. Run Report Stream locally with the AppInsights java agent and configured AppInsights environment variable
2. Submit a Report
3. Query for that report in the `traces` table in AppInsights
4. See the receiver name and report id as queryable fields in `customDimensions`

## Changes
- Structured logging
- Extension functions for adding MDC fields

### Testing

## Linked Issues
- Fixes #12722
